build/build.py

```diff
@@ -67,8 +67,33 @@ def apply_patch(patch):
             print(test["package"], "-run", testName)
             print(go.test(test["package"], "-run", testName))
 
+def parse_source(remote):
+    if remote == 'https://github.com/ethereum/go-ethereum':
+        return 'foundation'
+    elif remote == 'https://github.com/maticnetwork/bor/':
+        return 'bor'
+    elif remote == 'https://github.com/etclabscore/core-geth':
+        return 'etc'
+    else:
+        return 'unknown'
+
+
+def push_to_archive(archive, remote, tag, xplugeth_tag, xplugeth_branch):
+    git.remote.add("archive", archive)
```

Comment:
Pretty sure this line will throw an error if archive already exists, but the rest of what follows would still succeed.